### ISSUE TITLE
QueryString: array values with no indexing

### ISSUE THREAD

**Issue by vanodevium** [thephpleague/uri-src#144]
| Q       | A     |
|---------|-------|
| Package | components |
| Version | >=7.2 |

### Question

Is there any legal method to make repeatable parameters in query string for next configuration:

```
$params = [
    'a' => [1, 2, 3],
];

Modifier::from('https://awesome.url/')->mergeQueryParameters($params);
```

I want to receive:

`https://awesome.url/?a=1&a=2&a=3`

Thanx!

**Comment by nyamsprod** [comment]
@vanodevium there are multiple ways to achieve what you need:

### using a string

```php
Modifier::from('https://awesome.url/')
    ->appendQuery('a=1&a=2&a=3')
    ->getUriString();

//returns 'https://awesome.url/?a=1&a=2&a=3'
```

### using a list of pairs

```php
Modifier::from('https://awesome.url/')
    ->appendQueryPairs([
        ['a', 1],
        ['a', 2],
        ['a', 3],
    ])
    ->getUriString();

//returns 'https://awesome.url/?a=1&a=2&a=3'
```

### using something that ressembles PHP parsed query

```php
Modifier::from('https://awesome.url/')
    ->appendQueryParameters(['a' => 1])
    ->appendQueryParameters(['a' => 2])
    ->appendQueryParameters(['a' => 3])
    ->getUriString();

//returns 'https://awesome.url/?a=1&a=2&a=3'
```

You can get more information by visiting [the documentation website about query modifiers](https://uri.thephpleague.com/components/7.0/modifiers/#query-modifiers)

Query modifiers prefixed with `append` add the new values at end of the query while preserving the existing keys, their position and values whereas Query modifiers prefixed with `merge` will merge all parameters values so that each query name appears only once in the resulting query string and the last occurrence value takes precedence over all the values for a given parameter key. So in your case I would avoid the `merge*` methods.

**Comment by vanodevium** [comment]
@nyamsprod 

Sorry, you don't quite understand the question.

append* methods didn't resolve my case...

```php
$params = [
    'a' => [1, 2, 3],
];
```

This array is retrieved from the 3rd server as JSON, so the main question is:

Can I get as result repeatable parameters **WITHOUT** unpacking this array into pairs.

Something like

`$url->appendQueryParameters($params)` -> `https://awesome.url/?a=1&a=2&a=3`

**Comment by nyamsprod** [comment]
I did understand the question, but you need to update your query parameters to meet the methods 
 that I listed requirements. Either you use `array_map` or `array_reduce` before using one of the append methods. There's a reason why the parameters you used do not work because they inherently do not store the value places in a query string hence it is better to use query data as list of pairs 
TL;DR: query string are better reprensent like this:

```php
[
    ['a', 1],
    ['a', 2],
    ['a', 3],
]
```
because this representation preserve position and value, which is not the case with your array.
